_posts/2017-11-01-programming-but-in-which-language.md

```diff
@@ -0,0 +1,65 @@
+
+## _addApplication()_ - Everything in English 
+
+This is fine. We developers are used to reading and writing program code in English. If the code base, comments and even commit messages are totally in English it is even possible to have developers who do not speak Finnish in the team. That is very likely to happen even in the near future, because Solita is now recruiting also in Sweden. According to the comments from our developers, all code in English is the natural way to do it.
+
+Using English as the domain language is a natural choice if the customer operates internationally or the domain is already in English. If the customer gives the domain vocabulary, it is easy to adopt in the project and the conversation of business logic is not distracted by misunderstood semantics because of language.
+
+![code on screen](/img/programming-language/photo_koodia.JPG)
+
+If the domain can easily be translated into English, it can be done. This should be done carefully, maybe in cooperation with the customer. The translation might require some expertise, because it makes difference if you translate Finnish _asiantuntija_ literally as _thingKnower_ instead of more sensible word _expert_. After the translation, the vocabulary should be documented. Without shared vocabulary the code will soon be full of _experts_, _specialists_ and _professionals_ when every developer is making own translations. The documentation also helps new developers entering the project.
+
+If the domain is translated, the translation should be used in all layers. Keep your database, server, client and front consistent and use the same language and vocabulary in every level.
+
+## _addHakemus()_ - the Domain in Finnish
+
+But. Let us think that the customer is a Finnish company operating only in national business with a complex domain. Maybe the domain is even regulated by legislation. All the domain concepts might not even have an exactly equivalent word in English. There is no point of just making words up; it just leads to misunderstandings. When the lack of exact vocabulary can be seen as a risk, it might be reasonable to keep Finnish as the domain language. Using the legit domain vocabulary throughout the database, software, and documentation guarantees also a common language for the developers and the customer.
+
+Usually the decision of using the Finnish domain language is made with the customer. This, of course, makes difficult to adapt developers who do not speak Finnish to the project. One can surely learn the domain names, and if the code, comments and commit messages are in English, it might be possible to enter the project. However, in the real life, the method names can be quite complicated and impossible to understand despite of the few English words in the middle, or what would you say about actual test method called _kiinteistoWithEOWithVireillaolevaHaltijoidenSelvennysasiaHasOnlyVoimassaolevatHaltija_.
```

Comment:
"makes difficult to adapt" - "makes it difficult to adapt". Lisäksi adapt verbi hämää, mutta en keksi parempaakaan.